provider/image.go

```diff
@@ -642,3 +654,118 @@ func processLogLine(msg string) (string, error) {
 	}
 	return info, nil
 }
+
+func configureDockerClient(configs map[string]string) (*client.Client, error) {
+	// check for TLS inputs
+	var caMaterial, certMaterial, keyMaterial, certPath, host string
+	if val, ok := configs["caMaterial"]; ok {
+		caMaterial = val
+	}
+	if val, ok := configs["certMaterial"]; ok {
+		certMaterial = val
+	}
+	if val, ok := configs["keyMaterial"]; ok {
+		keyMaterial = val
+	}
+	if val, ok := configs["certPath"]; ok {
+		certPath = val
+	}
+	if val, ok := configs["host"]; ok {
+		host = val
+	}
+
+	if certMaterial != "" || keyMaterial != "" {
```

Comment:
Should this condition also be done over `caMaterial` since it's used after? Especially since the `dockerd` certificate should have been signed with that CA.